python/pyspark/pandas/series.py

```diff
@@ -944,6 +944,47 @@ def between(self, left: Any, right: Any, inclusive: bool = True) -> "Series":
 
         return lmask & rmask
 
+    def cov(self, other: "Series", min_periods: Optional[int] = None) -> float:
+        """
+        Compute covariance with Series, excluding missing values.
+
+        Parameters
+        ----------
+        other : Series
+            Series with which to compute the covariance.
+        min_periods : int, optional
+            Minimum number of observations needed to have a valid result.
+
+        Returns
+        -------
+        float
+            Covariance between Series and other
+
+        Examples
+        --------
+        >>> from pyspark.pandas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s1 = ps.Series([0.90010907, 0.13484424, 0.62036035])
+        >>> s2 = ps.Series([0.12528585, 0.26962463, 0.51111198])
+        >>> s1.cov(s2)
+        -0.016857626527158744
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
```

Comment:
Can we add the type checking logic for `other` and related tests ??

If `other` is not Series, we should manually catch it and raise `TypeError` as pandas does.

```python
>>> pser.cov([0.12528585, 0.26962463, 0.51111198])
Traceback (most recent call last):
...
TypeError: unsupported type: <class 'list'>
```